### PR TITLE
Update DG user stories, glossary, NFRs, use cases

### LINKED ISSUE #8: Update the DG: user stories, glossary, NFRs, use cases
Update the Developer Guide in `/docs` based on project notes

- target profiles, value proposition and user stories
- use cases
- non-functional requirements
- glossary: define terms that are worth recording

### PR DESCRIPTION
**Pull request by rebeccacxy**
Closes #8

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -268,60 +268,87 @@ _{Explain here how the data archiving feature will be implemented}_
...
+**Use case: UC01 - Delete ingredient**
 
 **MSS**
 
-1.  User requests to list persons
-2.  AddressBook shows a list of persons
-3.  User requests to delete a specific person in the list
-4.  AddressBook deletes the person
+1.  `User` requests to delete a specific ingredient by specifying its name, quantity and expiry date.
+2.  `Fridgy` deletes the ingredient.
```
Comment (rootkie) [inline docs/DeveloperGuide.md:305]: Should this be fully delete the item with matching name, quantity and expiry date only or can we adapt it to matching name and expiry date, then reduce the quantity. Delete item if quantity <= 0.

Comment (rebeccacxy) [reply docs/DeveloperGuide.md:305]: I was thinking `update` does the same job as change or reduce quantity. Delete can actually simply just `delete index` to fully delete?

Comment (rootkie) [reply docs/DeveloperGuide.md:305]: In that case we can just delete by name and expiry date as each ingredient can be uniquely identified by those 2 properties. Or we can display the index for user to delete.

Comment (rebeccacxy) [reply docs/DeveloperGuide.md]: Changes made

**Review by rootkie — APPROVED** [review]

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -351,19 +375,19 @@ testers are expected to do more *exploratory* testing.
...
 
-1. Deleting a person while all persons are being shown
+1. Deleting a recipe while all recipes are being shown
 
-   1. Prerequisites: List all persons using the `list` command. Multiple persons in the list.
+   1. Prerequisites: List all recipes using the `list` command. Multiple recipes in the list.
 
-   1. Test case: `delete 1`<br>
-      Expected: First contact is deleted from the list. Details of the deleted contact shown in the status message. Timestamp in the status bar is updated.
+   1. Test case: `delete recipe -n pasta`<br>
```
Comment (NayLin-H99) [inline docs/DeveloperGuide.md:384]: I think although we might enforce unique names for recipes, I believe we are still doing deletions by index?

**Review by NayLin-H99 — CHANGES_REQUESTED** [review]
LGTM apart from delete using names instead of index.

**Review by NayLin-H99 — APPROVED** [review]

**Pull request merged**